client.go

```diff
@@ -315,14 +318,28 @@ func (a *Client) CopyFromRemotePassThru(
 	remotePath string,
 	passThru PassThru,
 ) error {
+	_, err := a.CopyFromRemoteFileInfos(ctx, w, remotePath, passThru)
+
+	return err
+}
+
+// CopyFroRemoteFileInfos copies a file from the remote to a given writer and return a FileInfos struct
+// containing information about the file such as permissions, the file size, modification time and access time
+func (a *Client) CopyFromRemoteFileInfos(
+	ctx context.Context,
+	w io.Writer,
+	remotePath string,
+	passThru PassThru,
+) (*FileInfos, error) {
 	session, err := a.sshClient.NewSession()
 	if err != nil {
-		return fmt.Errorf("Error creating ssh session in copy from remote: %v", err)
+		return nil, fmt.Errorf("Error creating ssh session in copy from remote: %v", err)
 	}
 	defer session.Close()
 
 	wg := sync.WaitGroup{}
 	errCh := make(chan error, 4)
+	fileInfosCh := make(chan *FileInfos, 4)
```

Comment:
Can you moitvate the need for a `fileInfosCh`is it written to from multiple threads (goroutines) or can we simply assign to it? I don't think we need it for synchronisation either since we already use waitgroups for that?

Reply:
I changed accordingly.